lib/commands/deploynodeapp.js

```diff
@@ -104,25 +137,67 @@ module.exports.run = function(opts, cb) {
       }
     }
   }
-  descriptor.main.required = true;
+  if (opts.runtime != EdgeServicesRuntimeOption) {
+    descriptor.main.required = true;
+  }
+  
   descriptor.api.required = true;
   options.validateSync(opts, descriptor);
   if (opts.debug) {
     console.log('deploynodeapp: %j', opts);
   }
 
   var request = defaults.defaultRequest(opts);
-
   getDeploymentInfo(opts, request, function(err) {
     if (err) { return cb(err); }
 
-    // if preserve-policies, we do something entirely different...
-    if (opts['preserve-policies'] && opts.deploymentVersion > 1) {
-      return preservePoliciesRun(opts, cb);
-    }
+    var deploySeries;
+    if (opts.runtime === EdgeServicesRuntimeOption) {
+      // set defaults
+      opts['node-version'] = opts['node-version'] === undefined ? DefaultNodeVersion : opts['node-version'];
+      opts.edgeserviceshost = opts.edgeserviceshost === undefined ? 'https://apigee-interlude-gcpprod.e2e.apigee.net/turbo' : opts.edgeserviceshost;
+      opts.verbose = opts.verbose === undefined ? false : opts.verbose;
+      opts['base-path'] = (opts['base-path'] ? opts['base-path'] : '/'+opts.api);
+
+      opts['env-var'] = opts['env-var'] === undefined ? [] : opts['env-var'];
+      opts['config-var'] = opts['config-var'] === undefined ? [] : opts['config-var'];
```

Comment:
Any time you're using an `undefined` check _(this line and a few above)_ to use a default value, just use something like this:

```js
opts['config-var'] = opts['config-var'] || [];
```

I realize that there are cases where the first case can pass where you might not want it to but since we own this code, that seems unlikely and this is much simpler to read.  Better yet, use [underscore#defaults](http://underscorejs.org/#defaults).